File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/notice/NoticeActivity.java
package com.dhu.dhu_timetable.ui.navitem.notice;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


import com.dhu.dhu_timetable.R;
import com.dhu.dhu_timetable.model.NoticeModel;

import java.util.ArrayList;
import java.util.List;

public class NoticeActivity extends AppCompatActivity {
    private static final String TAG = "jaemin";

    private List<NoticeModel> noticeList = new ArrayList<>();
    private NoticeAdapter adapter;
    private RecyclerView recyclerView;
    private LinearLayoutManager layoutManager;

    private NoticeViewModel noticeViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notice);

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setTitle(getString(R.string.nav_notice_title));

        noticeViewModel = new ViewModelProvider(this).get(NoticeViewModel.class);
        noticeApi();

        recyclerView = findViewById(R.id.notice_recyclerview);
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        adapter = new NoticeAdapter(this, noticeViewModel);
        recyclerView.setAdapter(adapter);



        noticeViewModel.getNoticeData().observe(this, new Observer<List<NoticeModel>>() {
            @Override
            public void onChanged(List<NoticeModel> noticeModels) {
                if(noticeList != null){
                    noticeList = noticeModels;
                    adapter.setNoticeList(noticeModels);
                }
            }
        });

    }


    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch(item.getItemId()){
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void noticeApi() {
        noticeViewModel.setNoticeData();
    }


}

File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/notice/NoticeActivity.kt
package com.dhu.dhu_timetable.ui.navitem.notice

import android.os.Bundle
import android.view.MenuItem
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.SimpleItemAnimator
import com.dhu.dhu_timetable.R
import com.dhu.dhu_timetable.databinding.ActivityNoticeBinding
import java.util.*

class NoticeActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNoticeBinding

    private val noticeAdapter: NoticeAdapter by lazy { NoticeAdapter() }
    private val noticeViewModel: NoticeViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityNoticeBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar?.let {
            it.setDisplayHomeAsUpEnabled(true)
            it.title = getString(R.string.nav_notice_title)
        }

        noticeApi()
        setRecyclerView()

        noticeViewModel.noticeData.observe(this) { noticeModels ->
            noticeAdapter.setNoticeList(noticeModels)
        }
    }

    private fun setRecyclerView() {
        binding.noticeRecyclerview.apply {
            adapter = noticeAdapter.also { adapter ->
                adapter.setHasStableIds(true)
            }
            itemAnimator.also { animator ->
                when(animator) {
                    is SimpleItemAnimator -> animator.supportsChangeAnimations = false
                }
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun noticeApi() {
        noticeViewModel.setNoticeData()
    }

    companion object {
        private const val TAG = "jaemin"
    }
}